tests/components/homekit/test_type_security_systems.py

```diff
@@ -129,3 +136,116 @@ async def test_no_alarm_code(hass, hk_driver, config, events):
+                default_current_states["AwayArm"],
+            ],
+            "target_values": [
+                default_target_services["Disarm"],
+                default_target_services["AwayArm"],
+            ],
+        },
+        {
+            "features": SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY,
+            "current_values": [
+                default_current_states["Disarmed"],
+                default_current_states["AlarmTriggered"],
+                default_current_states["StayArm"],
+                default_current_states["AwayArm"],
+            ],
+            "target_values": [
+                default_target_services["Disarm"],
+                default_target_services["StayArm"],
+                default_target_services["AwayArm"],
+            ],
+        },
+        {
+            "features": SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_NIGHT,
+            "current_values": [
+                default_current_states["Disarmed"],
+                default_current_states["AlarmTriggered"],
+                default_current_states["StayArm"],
+                default_current_states["AwayArm"],
+                default_current_states["NightArm"],
+            ],
+            "target_values": [
+                default_target_services["Disarm"],
+                default_target_services["StayArm"],
+                default_target_services["AwayArm"],
+                default_target_services["NightArm"],
+            ],
+        },
+        {
+            "features": SUPPORT_ALARM_ARM_HOME
+            | SUPPORT_ALARM_ARM_AWAY
+            | SUPPORT_ALARM_ARM_NIGHT
+            | SUPPORT_ALARM_TRIGGER,
+            "current_values": [
+                default_current_states["Disarmed"],
+                default_current_states["AlarmTriggered"],
+                default_current_states["StayArm"],
+                default_current_states["AwayArm"],
+                default_current_states["NightArm"],
+            ],
+            "target_values": [
+                default_target_services["Disarm"],
+                default_target_services["StayArm"],
+                default_target_services["AwayArm"],
+                default_target_services["NightArm"],
+            ],
+        },
+    ]
+
+    for test_config in test_configs:
+        attrs = {"supported_features": test_config.get("features")}
+
+        hass.states.async_set(entity_id, None, attributes=attrs)
+        await hass.async_block_till_done()
+
+        acc = SecuritySystem(hass, hk_driver, "SecuritySystem", entity_id, 2, config)
+        await acc.run_handler()
+        await hass.async_block_till_done()
+
+        valid_current_values = acc.char_current_state.properties.get("ValidValues")
+        valid_target_values = acc.char_target_state.properties.get("ValidValues")
+
+        for val in valid_current_values.values():
+            assert val in test_config.get("current_values")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)